util/base_parser.h

```diff
@@ -42,15 +51,32 @@ class BaseParser {
     }
 
     constexpr void skip_whitespace() {
-        while (!is_eof() && is_space(peek())) {
+        while (!is_eof() && is_whitespace(peek())) {
             advance(1);
         }
     }
 
-private:
-    static constexpr auto space_chars = std::array{' ', '\f', '\n', '\r', '\t', '\v'};
-    constexpr static bool is_space(char c) { return std::ranges::find(space_chars, c) != end(space_chars); }
+    static constexpr bool is_whitespace(char c) {
+        return c == 0x20 || c == 0x0c || c == 0x0a || c == 0x0d || c == 0x09 || c == 0x0b;
```

Comment:
Why change these from `'\t'` and friends to ascii hex? imo it's more readable as `'\n'` when we're talking about printable characters anyway.

Especially for things like `is_upper_alpha`. Why not `return c >= 'A' && c <= 'Z';`?

Reply:
I was trying to keep it consistent. I can use escape characters instead for everything.

Reply:
Yes, please. Printable characters being readable without an ascii table would be nicer imo.